crates/oxc_linter/src/table.rs

```diff
@@ -106,54 +106,136 @@ impl RuleTable {
+            writeln!(s, "| {x:-<rule_width$} | {x:-<plugin_width$} | {x:-<7} | {x:-<8} |").unwrap();
+        }
 
         for row in rows {
             let rule_name = row.name;
             let plugin_name = &row.plugin;
+
+            let (enabled_mark, enabled_width) = if include_enabled {
+                if enabled.unwrap().contains(rule_name) {
+                    ("✅", ENABLED - 1)
+                } else {
+                    ("", ENABLED)
+                }
+            } else {
+                ("", ENABLED)
+            };
+
             let (default, default_width) =
                 if row.turned_on_by_default { ("✅", DEFAULT - 1) } else { ("", DEFAULT) };
             let rendered_name = if let Some(prefix) = link_prefix {
                 Cow::Owned(format!("[{rule_name}]({prefix}/{plugin_name}/{rule_name}.html)"))
             } else {
                 Cow::Borrowed(rule_name)
             };
-            let (fix_emoji, fix_emoji_width) = row.autofix.emoji().map_or(("", FIX), |emoji| {
-                let len = emoji.len();
-                if len > FIX { (emoji, 0) } else { (emoji, FIX - len) }
-            });
-            writeln!(s, "| {rendered_name:<rule_width$} | {plugin_name:<plugin_width$} | {default:<default_width$} | {fix_emoji:<fix_emoji_width$} |").unwrap();
+            // Improved mapping for emoji column alignment, allowing FIX to grow for negative display widths
+            let (fix_emoji, fix_emoji_width) =
+                Self::calculate_fix_emoji_width(row.autofix.emoji(), FIX);
+            if include_enabled {
+                writeln!(
+                    s,
+                    "| {rendered_name:<rule_width$} | {plugin_name:<plugin_width$} | {default:<default_width$} | {enabled_mark:<enabled_width$} | {fix_emoji:<fix_emoji_width$} |"
+                )
+                .unwrap();
+            } else {
+                writeln!(
+                    s,
+                    "| {rendered_name:<rule_width$} | {plugin_name:<plugin_width$} | {default:<default_width$} | {fix_emoji:<fix_emoji_width$} |"
+                )
+                .unwrap();
+            }
         }
 
         s
     }
+
+    /// Calculate the width adjustment needed for emoji fixability indicators
+    #[expect(clippy::cast_sign_loss)]
+    fn calculate_fix_emoji_width(emoji: Option<&str>, fix_col_width: usize) -> (&str, usize) {
+        emoji.map_or(("", fix_col_width), |emoji| {
+            let display_width: isize = match emoji {
+                "⚠️🛠️️" => -3,
+                "⚠️🛠️️💡" => -2,
+                "🛠️" => -1,
+                "💡" | "🚧" => 1,
+                "" | "🛠️💡" | "⚠️💡" => 0,
+                _ => 2,
+            };
+            let width = if display_width < 0 {
+                fix_col_width.saturating_add((-display_width) as usize)
+            } else {
+                fix_col_width.saturating_sub(display_width as usize)
+            };
+            (emoji, width)
+        })
+    }
```

Comment:
A bit cursed but I don't have a great alternative solution, and the rule table misalignment was bothering me before.